src/Visualization/Python/PlotTrajectories.py

```diff
@@ -0,0 +1,158 @@
+    show_or_save_plot_command,
+)
+
+
+@click.command(name="trajectories")
+@click.argument(
+    "h5_files",
+    nargs=-1,
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+)
+@click.option(
+    "--subfile-name-aha",
+    "-A",
+    default="ApparentHorizons/ControlSystemAhA_Centers.dat",
+    help=(
+        "Name of subfile containing the apparent horizon centers for object A."
+    ),
+)
+@click.option(
+    "--subfile-name-ahb",
+    "-B",
+    default="ApparentHorizons/ControlSystemAhB_Centers.dat",
+    help=(
+        "Name of subfile containing the apparent horizon centers for object B."
+    ),
+)
+@show_or_save_plot_command()
+@apply_stylesheet_command()
+def plot_trajectories_command(h5_files, subfile_name_aha, subfile_name_ahb):
+    """Plot trajectories in inspiral simulation
+
+    Concatenates partial trajectories from each h5 reductions files and plots
+    full trajectories.
+    """
+    reductionfiles = h5_files
+    AhA, AhB = import_A_and_B(
+        reductionfiles, subfile_name_aha, subfile_name_ahb
+    )
+    if AhA is not None and AhB is not None:
+        plot_trajectory(AhA, AhB)
+    else:
+        click.echo("Error: Unable to import data or not valid data.")
+
+
+def import_A_and_B(filenames, subfile_name_aha, subfile_name_ahb):
+    A_data = []
+    B_data = []
+
+    for filename in filenames:
+        with h5py.File(filename, "r") as file:
+            if subfile_name_aha in file:
+                A_data.append(np.array(file[subfile_name_aha][:]))
+            if subfile_name_ahb in file:
+                B_data.append(np.array(file[subfile_name_ahb][:]))
+
+    if A_data and B_data:
+        A = np.concatenate(A_data)
+        B = np.concatenate(B_data)
+        min_length = min(len(A), len(B))
+        return A[:min_length], B[:min_length]
+    else:
+        return None, None
+
+
+def plot_trajectory(AhA, AhB, sample_rate=15):
+    indexs = [4, 5, 6]  # 4 -> x, 5 -> y, 6 -> z
+
+    fig = plt.figure(figsize=(10, 10))
+
+    # Sample data for 3D plots
+    AhA_sampled = AhA[::sample_rate]
+    AhB_sampled = AhB[::sample_rate]
+
+    # Plot 3D trajectories
+    ax1 = fig.add_subplot(2, 2, 1, projection="3d")
+    ax1.scatter(
+        AhA_sampled[:, indexs[0]],
+        AhA_sampled[:, indexs[1]],
+        AhA_sampled[:, indexs[2]],
+        color="blue",
+        label="AhA",
+        s=2,
+    )
+    ax1.scatter(
+        AhB_sampled[:, indexs[0]],
+        AhB_sampled[:, indexs[1]],
+        AhB_sampled[:, indexs[2]],
+        color="C1",
+        label="AhB",
+        s=2,
+    )
+    ax1.set_xlabel("X")
+    ax1.set_ylabel("Y")
+    ax1.set_zlabel("Z")
+    ax1.set_title("3D Trajectories")
+    ax1.legend()
+
+    # Calculate coordinate separation in 3D
+    x_diff_3d = AhA_sampled[:, indexs[0]] - AhB_sampled[:, indexs[0]]
+    y_diff_3d = AhA_sampled[:, indexs[1]] - AhB_sampled[:, indexs[1]]
+    z_diff_3d = AhA_sampled[:, indexs[2]] - AhB_sampled[:, indexs[2]]
```

Comment:
This is just:

```
separation_3d = AhA_sampled[:, 1:] - AhB_sampled[:, 1:]
```

This computes all 3 components at once. For 2d it's `[:, 1:3]`.